src/background.ts

```diff
@@ -138,6 +138,31 @@ browser.tabs.onActivated.addListener(ev => {
         .catch(ignore)
 })
 
+// Valid events listed here: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest#Events
+for (const requestEvent of [
+    "AuthRequired",
+    "BeforeRedirect",
+    "BeforeRequest",
+    "BeforeSendHeaders",
+    "Completed",
+    "ErrorOccured",
+    "HeadersReceived",
+    "ResponseStarted",
+    "SendHeaders",
+]) {
+    config.getAsync("autocmds", requestEvent).then(aucmds => {
+        if (!aucmds) return
+        const patterns = Object.keys(aucmds)
+        for (const pattern of patterns) {
```

Comment:
This should be moved out into a `registerWebRequestAutocmd` so that we can add a config listener that calls it (on a single event + pattern) every time a webrequest auto command is added.

It would be nice if we could deal with removal of web request aucmds too.